### ISSUE TITLE
Comma on newline preceded by // comment will move code into comment

### ISSUE THREAD

**Issue by liao123abc** [pinterest/ktlint#367]
I got a when phrase. The case is

before
```java
when (type) {
            IHomeLivingConstant.LIVING_TYPE_GUESS_YOU_LIKE //a
                , IHomeLivingConstant.Live_MODULE_CONTENT_CODE // b
                , IHomeLivingConstant.LIVE_MODULE_AUTO_PLAY  //c
-> {
    // do something
}
```

after formatting, I turn out
```java
when (type) {
            IHomeLivingConstant.LIVING_TYPE_GUESS_YOU_LIKE //a, IHomeLivingConstant.Live_MODULE_CONTENT_CODE // b, IHomeLivingConstant.LIVE_MODULE_AUTO_PLAY  //c
-> {
    //do something
} 

```

This lead to bugs cause it will not invoke compile error, but is wrong logically.  And it is hard to find the bug in the hundreds of formatted files.

**Comment by shashachu** [comment]
It seems difficult for ktlint to autoformat correctly in this case, but it does seem like it should be a linter error. The Kotlin Android style guide says:
`A comma (,) stays attached to the token that precedes it.` as well as `A lambda arrow (->) stays attached to the argument list that precedes it.`

I'm not familiar enough yet with the code to know which rule is reformatting the code like this.
